custom_components/elasticsearch/es_index_manager.py

```diff
@@ -194,55 +219,48 @@ async def _create_legacy_template(self):
                 )
             except ElasticsearchException as err:
                 LOGGER.exception("Error creating initial index/alias: %s", err)
-        elif self._using_ilm:
-            LOGGER.debug("Ensuring ILM Policy is attached to existing index")
-            try:
-                await client.indices.put_settings(
-                    index=self.index_alias,
-                    preserve_existing=True,
-                    body={
-                        "index.lifecycle.name": self._ilm_policy_name,
-                        "index.lifecycle.rollover_alias": self.index_alias,
-                    },
-                )
-            except ElasticsearchException as err:
-                LOGGER.exception("Error updating index ILM settings: %s", err)
 
-    async def _create_ilm_policy(self, config):
+    async def _create_basic_ilm_policy(self, ilm_policy_name):
         """Create the index lifecycle management policy."""
         from elasticsearch7.exceptions import TransportError
 
         client = self._gateway.get_client()
 
         try:
-            existing_policy = await client.ilm.get_lifecycle(self._ilm_policy_name)
+            existing_policy = await client.ilm.get_lifecycle(ilm_policy_name)
         except TransportError as err:
             if err.status_code == 404:
                 existing_policy = None
             else:
                 LOGGER.exception("Error checking for existing ILM policy: %s", err)
                 raise err
```

Comment:
I know it's not a part of this PR, but can we raise one of our own errors instead?